sync/downstream.py

```diff
@@ -97,22 +107,53 @@ def pr_status(self, value):
         self.data["pr-status"] = value
 
     @property
-    def metadata_ready(self):
-        # This is mostly for testing
+    def next_action(self):
+        """Work out the next action for the sync based on the current status.
+
+        Returns a DownstreamAction indicating the next step to take."""
+
         if self.data.get("force-metadata-ready"):
-            return True
+            # This is mostly for testing
+            return DownstreamAction.ready
         if self.skip:
-            return True
+            return DownstreamAction.ready
         if not self.requires_try:
-            return True
+            return DownstreamAction.ready
+        if self.error:
+            return DownstreamAction.manual_fix
+
         latest_try_push = self.latest_valid_try_push
+
         if not latest_try_push:
-            return False
-        if latest_try_push.status != "complete" or latest_try_push.infra_fail:
-            return False
-        if self.affected_tests() and latest_try_push.expired():
-            return False
-        return latest_try_push.stability or not self.requires_stability_try
+            return DownstreamAction.try_push
+
+        if latest_try_push.status != "complete":
+            return DownstreamAction.wait_try
+        if latest_try_push.infra_fail:
+            if len(self.latest_busted_try_pushes()) > 5:
+                message = ("Too many busted try pushes. "
+                           "Check the try results for infrastructure issues.")
+                self.error = message
```

Comment:
We don't want to comment on the bug here?

Reply:
So, the reason I don't do that is that we don't have any way to know if the comment was already made. We probably do want some notification on the bug though.